source/geod24/LocalRest.d

```diff
@@ -212,75 +199,216 @@ private struct ArgWrapper (T...)
     T args;
 }
 
-/// Our own little scheduler
-private final class LocalScheduler : C.FiberScheduler
+// very simple & limited variant, to keep it performant.
+// should be replaced by a real Variant later
+private struct Variant
 {
-    import core.sync.condition;
-    import core.sync.mutex;
+    this (Command msg) { this.cmd = msg; this.tag = Variant.Type.command; }
+    this (Response msg) { this.res = msg; this.tag = Variant.Type.response; }
+    this (FilterAPI msg) { this.filter = msg; this.tag = Variant.Type.filter; }
+    this (TimeCommand msg) { this.time = msg; this.tag = Variant.Type.timeCommand; }
+    this (ShutdownCommand msg) { this.shutdown = msg; this.tag = Variant.Type.shutdownCommand; }
 
-    /// Just a FiberBinarySemaphore with a state
-    private struct Waiting { FiberBinarySemaphore s; bool busy; }
+    union
+    {
+        Command cmd;
+        Response res;
+        FilterAPI filter;
+        TimeCommand time;
+        ShutdownCommand shutdown;
+    }
 
-    /// The 'Response' we are currently processing, if any
-    private Response pending;
+    Type tag;
 
-    /// Request IDs waiting for a response
-    private Waiting[ID] waiting;
+    /// Type of a request
+    enum Type : ubyte
+    {
+        command,
+        response,
+        filter,
+        timeCommand,
+        shutdownCommand,
+    }
+}
 
-    /// scheduler ID
-    private size_t sched_id;
+private alias CommandChn = C.Channel!Variant;
+private alias RespChn = C.Channel!Response;
 
-    /// Initialize this scheduler and its unique ID
-    public this () @safe @nogc nothrow
+/// Represents a connection between a server and a client
+private class Connection
+{
+    ///
+    this () @trusted nothrow
     {
-        static size_t last_idx;
-        this.sched_id = last_idx++;
+        this.command_chn = new CommandChn();
+        this.resp_chn = new RespChn();
     }
```

Comment:
I don't see any reason this should be `@trusted`. Either the ctors are `@safe` / `@trusted`, or this is `@system`.
`@trusted` would only make sense if we were passing parameters, but here we're not.

Reply:
Yeah you are right, I might have been too generous with my `@trusted` usage.